core/types/subtyping.cc

```diff
@@ -1259,7 +1265,13 @@ bool isSubTypeUnderConstraintSingle(const GlobalState &gs, TypeConstraint &const
                             ++i;
                             result = Types::isSubTypeUnderConstraint(gs, constr, a1.elems[i], el2, mode);
```

Comment:
To prevent this sort of mistake from having happened in the first place, I would actually make it so that the only version of `isSubType` that allows omitting parameters is the one that just takes gs and two types. I.e., if you're passing in a mode, you should be forced to pass in a collector too.

Reply:
I looked through a bunch of the uses of `isSubType` (and variants) that don't pass in a mode, and a bunch of those are used for error reporting too, so I think I'll just make `errorCollector` required for every call for now.